jdaviz/__init__.py

```diff
@@ -26,14 +26,92 @@
            'viewers',
            'new_viewers',
            'get_data']
-_incl = ['App', 'enable_hot_reloading', '__version__']
+_incl = ['enable_hot_reloading', '__version__', 'gca', 'get_all_apps', 'app']
 _temporary_incl = ['open', 'Cubeviz', 'Imviz', 'Mosviz', 'Rampviz', 'Specviz', 'Specviz2d']
 __all__ = _expose + _incl + _temporary_incl
 
 
-global _ca
+global _apps
+global _current_index
 
-_ca = App(api_hints_obj='jd')
+_apps = []
+
+
+def app(replace=False, set_as_current=True):
+    """
+    Create a new jdaviz application instance and assign as the current instance.
+
+    Parameters
+    ----------
+    replace : bool, optional
+        If True, replaces the current application instance with the new one.
+        Default is False, which means a new instance is added to the list of applications.
+    set_as_current : bool, optional
+        If True, sets the newly created application instance as the current instance.
+        Default is True.
+
+    Returns
+    -------
+    App
+        A new instance of the App class.
+    """
+    global _apps
+    global _current_index
+    ca = App(api_hints_obj='jd')
+    if replace:
+        _apps[_current_index] = ca
+    else:
+        _apps += [ca]
+        if set_as_current:
+            _current_index = len(_apps) - 1
+    return ca
+
+
+def get_all_apps():
+    """
+    Get a list of all jdaviz application instances.
+
+    Returns
+    -------
+    list
+        A list of all jdaviz application instances.
+    """
+    return _apps
+
+
+def gca(index=None, set_as_current=True):
```

Comment:
I feel slightly silly asking this, but can you remind me what "gca" means? And given that it's not obvious, maybe this should have a more descriptive name? My guess would be "get current app". I'm generally in favor of more descriptive method names rather than opaque abbreviations, especially when I can't imagine this is something that would get called constantly.

Reply:
It does mean "get current app", yes.  The thinking was to parallel matplotlib's syntax (as that motivates the entire concept of being able to act on the most recent instance) which uses `gcf()` (get current figure) and `gca()` (get current axis).